src/modules/transaction/components/ExportCSVComponent.vue

```diff
@@ -214,17 +241,29 @@ let loadTransactions = async () => {
       }
     }
   }
-  if(startBlock.value && endBlock.value){
-    txnQueryParams.fromHeight = startBlock.value;
-    txnQueryParams.toHeight = endBlock.value;
+  if(option.value === "blockNumber"){
+    if(startBlock.value && endBlock.value){
+      txnQueryParams.fromHeight = startBlock.value;
+      txnQueryParams.toHeight = endBlock.value;
+    }
+  }
+  else{
+    if(startBlock.value){
+      const chainConfig = new ChainProfileConfig(networkState.chainNetworkName);
+      chainConfig.init();
+      let blockTargetTime = parseInt(chainConfig.blockGenerationTargetTime);
+      let blockTargetTimeByDay = (60 * 60 * 24) / blockTargetTime;
+      endBlock.value = await AppState.chainAPI.chainAPI.getBlockchainHeight()
+      txnQueryParams.fromHeight = endBlock.value -  Math.floor(startBlock.value * blockTargetTimeByDay)
```

Comment:
please use another variable instead of startBlock, check for possibility of negative value